src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs

```diff
@@ -43,7 +43,7 @@ private bool TryLoadManagedAssemblyMetadata()
             try
             {
                 // Try to load the file using the managed metadata reader
-                using (FileStream assemblyStream = File.OpenRead(_fileName))
+                using (FileStream assemblyStream = new FileStream(_fileName, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 0x1000, useAsync: false))
```

Comment:
Common block size, common page size and exceptions tend to be small whole multiples. I did once profile whether 4KiB really did work out for the best buffer size for streams or if it was just a habit we'd all fallen into because it seemed sensible. It was slightly better for network streams and a lot better for file streams.